src/Channels.Networking.Windows.RIO/Internal/RioThread.cs

```diff
@@ -130,16 +139,50 @@ private static void OnThreadStart(object state)
                     var error = GetLastError();
                     if (error != 258)
                     {
-                        throw new Exception(string.Format("ERROR: GetQueuedCompletionStatusEx returned {0}", error));
+                        throw new Exception($"ERROR: GetQueuedCompletionStatusEx returned {error}");
                     }
                 }
             }
+
+#if net451
+            Thread.EndThreadAffinity();
+#endif
         }
 
         [DllImport(Kernel_32, SetLastError = true)]
         private static extern bool GetQueuedCompletionStatus(IntPtr CompletionPort, out uint lpNumberOfBytes, out uint lpCompletionKey, out NativeOverlapped* lpOverlapped, int dwMilliseconds);
 
         [DllImport(Kernel_32, SetLastError = true)]
         private static extern long GetLastError();
+
+        [DllImport("kernel32.dll")]
+        private static extern int GetCurrentThreadId();
+
+        private static ProcessThread GetCurrentThread()
+        {
+            var id = GetCurrentThreadId();
+            var processThreads = Process.GetCurrentProcess().Threads;
+
+            for (var i = 0; i < processThreads.Count; i++)
+            {
+                var thread = processThreads[i];
+                if (thread.Id == id)
+                {
+                    return thread;
+                }
+            }
+
+            return null;
+        }
+
+        private static int GetAffinity(int i, int cores)
```

Comment:
Probably should create a physical core mask and use that rather than assuming its every other one.